### PR TITLE
macOS: split linting & testing

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR adds a new job that's solely responsible for macos code linting

### Motivation

This will hopefully improve our macOS jobs duration by running them in parallel.  https://datadoghq.atlassian.net/browse/APL-2393

### Additional Notes

This is based on top of #20856 and the commits from this PR are also included here until the freeze is lifted

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/github_tasks.py [hunk 1]
```diff
@@ -66,28 +80,36 @@ def trigger_macos_test(
...
+    datadog_agent_ref=DEFAULT_BRANCH,
+    release_version="nightly-a7",
+    python_runtimes="3",
+    version_cache=None,
+):
+    _trigger_macos_workflow(
+        release_version,
+        None,
+        0,
+        0,
```
Comment (Pythyu) [inline tasks/github_tasks.py:107]: ❓ Question :
Maybe worth putting these args as default value ?

**Review by Pythyu — APPROVED** [review]
Otherwise LGTM

File: tasks/github_tasks.py (hunk 1 above)
Comment (chouquette) [reply tasks/github_tasks.py:107]: I wasn't sure how mixing default values & kwargs would behave but on a second thought I don't really see what the problem could be, I'll make the change

**Review by hush-hush — APPROVED** [review]

**Pull request merged**